packages/shared/src/hooks/useTimedAnimation.ts

```diff
@@ -1,80 +1,81 @@
   autoEndAnimation: boolean;
-  animationDuration: number;
+  outAnimationDuration?: number;
   onAnimationEnd?: () => void;
-  onAnimationStop?: () => void;
 }
 
-const TEMPORARY_ID = 1;
-const INTERVAL_COUNT = 10;
-const IN_OUT_ANIMATION = 140;
+const PROGRESS_INTERVAL = 10;
+const OUT_ANIMATION_DURATION = 140;
+const MANUAL_DISMISS_ANIMATION_ID = 1;
 
 export const useTimedAnimation = ({
   autoEndAnimation = true,
-  animationDuration,
+  outAnimationDuration = OUT_ANIMATION_DURATION,
   onAnimationEnd,
-  onAnimationStop,
 }: UseTimedAnimationProps): UseTimedAnimation => {
   const timeout = useRef<number>();
   const [timer, setTimer] = useState(0);
-  const [intervalId, setIntervalId] = useState<number>(null);
+  const interval = useRef<number>();
 
-  const endAnimation = useCallback(() => {
-    if (!isNullOrUndefined(animationDuration)) {
+  const clearInterval = () => {
+    if (!interval?.current) {
       return;
     }
 
-    window.clearInterval(intervalId);
-    onAnimationStop();
-  }, [animationDuration, intervalId]);
+    window.clearInterval(interval.current);
+    interval.current = null;
+  };
 
-  useEffect(() => {
-    // when the timer ends we need to do cleanups
-    // we delay the callback execution so we can let the slide out animation finish
-    if (!isNullOrUndefined(animationDuration) && timer === 0 && intervalId) {
-      window.clearInterval(intervalId);
-      setIntervalId(null);
-      window.clearTimeout(timeout.current);
-      timeout.current = window.setTimeout(onAnimationEnd, IN_OUT_ANIMATION);
-    }
-  }, [timer, animationDuration, intervalId]);
-
-  useEffect(() => {
-    if (isNullOrUndefined(animationDuration)) {
+  const endAnimation = () => {
+    if (!interval.current) {
       return;
     }
 
-    window.clearInterval(intervalId);
-    setTimer(animationDuration);
+    setTimer(0);
+  };
 
-    if (!autoEndAnimation && !intervalId) {
-      setIntervalId(TEMPORARY_ID);
+  const startAnimation = (duration: number) => {
+    if (isNullOrUndefined(duration) || duration <= 0) {
       return;
     }
 
-    if (animationDuration <= 0) {
+    setTimer(duration);
+
+    if (!autoEndAnimation) {
+      clearInterval();
+      interval.current = MANUAL_DISMISS_ANIMATION_ID;
       return;
     }
 
-    setIntervalId(
-      window.setInterval(
-        () =>
-          setTimer((current) =>
-            INTERVAL_COUNT >= current ? 0 : current - INTERVAL_COUNT,
-          ),
-        INTERVAL_COUNT,
-      ),
+    clearInterval();
+    interval.current = window.setInterval(
+      () =>
+        setTimer((current) =>
+          PROGRESS_INTERVAL >= current ? 0 : current - PROGRESS_INTERVAL,
+        ),
+      PROGRESS_INTERVAL,
     );
-  }, [animationDuration]);
+  };
+
+  useEffect(() => {
+    // when the timer ends we need to do cleanups
+    // we delay the callback execution so we can let the slide out animation finish
+    if (timer === 0 && interval?.current) {
```

Comment:
To make sure we don't introduce infinite interval

```suggestion
    if (timer <= 0 && interval?.current) {
```